### PR TITLE
Migrate helm plugin to use dynamic extensions

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Fixes**: https://issues.redhat.com/browse/ODC-6198

**Analysis / Root cause**: Helm plugin is still using static extensions for its contributions.

**Solution Description**: 
- Migrate all static contributions to dynamic ones.
- Add missing support for dynamic topology options extension.

**Screen shots / Gifs for design review**: No UI change.

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge

### CODE REVIEW

File: frontend/packages/helm-plugin/src/plugin.ts [hunk 1]
```diff
@@ -1,125 +1,3 @@
...
-          )
-        ).default,
-    },
-    flags: {
-      required: [FLAG_OPENSHIFT_HELM],
-    },
-  },
-  ...helmTopologyPlugin,
-];
+const plugin = [];
```
Comment (invincibleJai) [inline frontend/packages/helm-plugin/src/plugin.ts:1]: Do we still need this?

File: frontend/packages/helm-plugin/console-extensions.json [hunk 2]
```diff
@@ -1,4 +1,89 @@
...
+  {
+    "type": "console.resource-metadata",
+    "properties": {
+      "model": {
+        "group": "helm.openshift.io",
+        "version": "v1beta1",
+        "kind": "HelmChartRepository"
+      },
+      "label": "Helm Chart Repository",
+      "labelPlural": "Helm Chart Repositories",
```
Comment (invincibleJai) [inline frontend/packages/helm-plugin/console-extensions.json:11]: do we need i18n support here?

```suggestion
      "label": "%helm~Helm Chart Repository%",
      "labelPlural": "%helm~Helm Chart Repositories%",
```

File: frontend/packages/helm-plugin/src/plugin.ts (hunk 1 above)
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/src/plugin.ts:1]: I don't think we need this but that's the way I see kubevirt plugin is doing. I think its okay to have it here.

File: frontend/packages/helm-plugin/console-extensions.json (hunk 2 above)
Comment (rohitkrai03) [reply frontend/packages/helm-plugin/console-extensions.json:11]: Do we actually translate our models?

Comment (rohitkrai03) [reply frontend/packages/helm-plugin/console-extensions.json]: Done.

**Comment by invincibleJai** [comment]
/label qe-approved
/label docs-approved
/label px-approved

added docs-approved on behalf of the docs team as the epic has a doc-ack label.
added px-approved on behalf of the px team as the epic has a px-ack label.
Epic link: https://issues.redhat.com/browse/ODC-5712

**Pull request merged**